docs/user_guides/langchain-integration.md

```diff
@@ -0,0 +1,167 @@
+# LangChain Integration
+
+This guide will teach you how to integrate guardrail configurations built with NeMo Guardrails into your LangChain applications. The examples in this guide will focus on using the [LangChain Expression Language](https://python.langchain.com/docs/expression_language/) (LCEL).
+
+## Overview
+
+NeMo Guardrails provides a LangChain native interface that implements the [Runnable Protocol](https://python.langchain.com/docs/expression_language/interface), through the `RunnableRails` class. To get started, you must first load a guardrail configuration and create a `RunnableRails` instance:
+
+```python
+from nemoguardrails import RailsConfig
+from nemoguardrails.integrations.langchain.runnable_rails import RunnableRails
+
+config = RailsConfig.from_path("path/to/config")
+guardrails = RunnableRails(config)
+```
+
+To add guardrails around an LLM model inside a chain, you have to "wrap" the LLM model with a `RunnableRails` instance, i.e., `(guardrails | ...)`.
+
+Let's take a typical example using a prompt, a model, and an output parser:
+
+```python
+from langchain.chat_models import ChatOpenAI
+from langchain.prompts import ChatPromptTemplate
+from langchain_core.output_parsers import StrOutputParser
+
+prompt = ChatPromptTemplate.from_template("tell me a short joke about {topic}")
+model = ChatOpenAI()
+output_parser = StrOutputParser()
+
+chain = prompt | model | output_parser
+```
+
+To add guardrails around the LLM model in the above example:
+
+```python
+chain_with_guardrails = prompt | (guardrails | model) | output_parser
+```
+> **NOTE**: Using the extra parenthesis is essential to enforce the order in which the `|` (pipe) operator is applied.
```

Comment:
This might be a sign it's not implemented properly. Ideally sequences should pass data forward, in which case nesting the sequences shouldn't matter. Will look at code in a sec!